src/main/java/com/arpnetworking/commons/test/ThreadLocalBuildableTestHelper.java

```diff
@@ -119,7 +119,17 @@
                             field.getName()),
                     providedValue,
                     resetBuilderValue);
-            Assert.assertSame(
+            // It would be better for this next one to be `assertSame()` rather than `assertEquals()`,
+            //   since that would require default field values to be reused, thereby discouraging `reset()`
+            //   from creating new objects.
+            // However: consider JSON deserialization with a @JsonAnySetter.
+            //   Such a method might get called many, many times, and we don't want to build a new ImmutableMap each time,
```

Comment:
So.... if the map is immutable, could we not just have a singleton empty immutable map?